### PR TITLE
Refinance configure flow sidebar

### PR DESCRIPTION
**Pull request by piekczyk**
# [Refinance configure flow sidebar](https://app.shortcut.com/oazo-apps/story/14936/create-or-re-use-dpm-for-the-user)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- configured flow sidebar with filters per given protocol and skeleton when sidebar data is being loaded
  
## How to test 🧪
  <Please explain how to test your changes>

- dpm proxies should be handled properly based on its availability and provided filter

### CODE REVIEW

File: features/refinance/contexts/RefinanceGeneralContext.tsx [hunk 1]
```diff
@@ -43,11 +49,14 @@ export type RefinanceContextInput = {
...
     debtTokenSymbol: string
     maxLtv: RiskRatio
+    pairId: number
   }
   environment: {
     tokenPrices: Record<string, string>
     chainId: number
     slippage: number
+    protocol: LendingProtocol
+    productType: OmniProductType
```
Comment (piotrwitek) [inline features/refinance/contexts/RefinanceGeneralContext.tsx:59]: protocol & type is on the position

File: features/refinance/hooks/useMakerRefinanceContextInputs.ts [hunk 2]
```diff
@@ -85,6 +89,7 @@ export const useMakerRefinanceContextInputs = ({
     collateralPrice,
     debtPrice: '1',
     poolId,
+    pairId: 1,
```
Comment (piotrwitek) [inline features/refinance/hooks/useMakerRefinanceContextInputs.ts:92]: what does it mean pairId: 1?

File: features/refinance/hooks/useMakerRefinanceContextInputs.ts [hunk 3]
```diff
@@ -96,5 +101,7 @@ export const useMakerRefinanceContextInputs = ({
     maxLtv,
     automations,
     contextId: id,
+    protocol: LendingProtocol.Maker,
```
Comment (piotrwitek) [inline features/refinance/hooks/useMakerRefinanceContextInputs.ts:104]: duplication, protocol is already on position

File: features/refinance/state/refinanceFormReducto.types.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export interface RefinanceFormState {
   refinanceOption?: RefinanceOptions
   strategy?: ProductHubItem // strip to bare minimum during clean up
   dpmProxy?: string
+  hasSimilarPosition?: boolean
```
Comment (piotrwitek) [inline features/refinance/state/refinanceFormReducto.types.ts:10]: what does it mean?

**Review by piotrwitek — CHANGES_REQUESTED** [review]
Great work added few comments and clarification questions.

File: features/refinance/contexts/RefinanceGeneralContext.tsx (hunk 1 above)
Comment (piekczyk) [reply features/refinance/contexts/RefinanceGeneralContext.tsx:59]: adjusted

File: features/refinance/hooks/useMakerRefinanceContextInputs.ts (hunk 2 above)
Comment (piekczyk) [reply features/refinance/hooks/useMakerRefinanceContextInputs.ts:92]: recently we updated url structure so with handling of different pools per specific tokens pair, so for example there may be a ETH/USDC-1 or ETH/USDC-2

File: features/refinance/hooks/useMakerRefinanceContextInputs.ts (hunk 3 above)
Comment (piekczyk) [reply features/refinance/hooks/useMakerRefinanceContextInputs.ts:104]: it's argument passed to get refinance ctx input, I don't see a duplication here

File: features/refinance/state/refinanceFormReducto.types.ts (hunk 4 above)
Comment (piekczyk) [reply features/refinance/state/refinanceFormReducto.types.ts:10]: it means that user already have position like the one to which he tries to refinance

**Review by piotrwitek — APPROVED** [review]
All comments were addressed looking good

**Pull request merged**